examples/controlnet/train_controlnet_flux.py

```diff
@@ -0,0 +1,1416 @@
+            sigma = sigma.unsqueeze(-1)
+        return sigma
+    
+    image_logs = None
+    for epoch in range(first_epoch, args.num_train_epochs):
+        for step, batch in enumerate(train_dataloader):
+            with accelerator.accumulate(flux_controlnet):
+                # Convert images to latent space
+                # vae encode
+                pixel_values = batch["pixel_values"].to(dtype=weight_dtype)
+                pixel_latents_tmp = vae.encode(pixel_values).latent_dist.sample()
+                pixel_latents_tmp = (pixel_latents_tmp - vae.config.shift_factor) * vae.config.scaling_factor
+                pixel_latents = FluxControlNetPipeline._pack_latents(
+                    pixel_latents_tmp,
+                    pixel_values.shape[0],
+                    pixel_latents_tmp.shape[1],
+                    pixel_latents_tmp.shape[2],
+                    pixel_latents_tmp.shape[3],
+                )
+
+                control_values = batch["conditioning_pixel_values"].to(dtype=weight_dtype)
+                control_latents = vae.encode(control_values).latent_dist.sample()
+                control_latents = (control_latents - vae.config.shift_factor) * vae.config.scaling_factor
+                control_image = FluxControlNetPipeline._pack_latents(
+                    control_latents,
+                    control_values.shape[0],
+                    control_latents.shape[1],
+                    control_latents.shape[2],
+                    control_latents.shape[3],
+                )
+
+                latent_image_ids = FluxControlNetPipeline._prepare_latent_image_ids(
+                    batch_size=pixel_latents_tmp.shape[0],
+                    height=pixel_latents_tmp.shape[2],
+                    width=pixel_latents_tmp.shape[3],
+                    device=pixel_values.device,
+                    dtype=pixel_values.dtype,
+                )
+
+                bsz = pixel_latents.shape[0]
+                noise = torch.randn_like(pixel_latents).to(accelerator.device).to(dtype=weight_dtype)
+                # Sample a random timestep for each image
+                # for weighting schemes where we sample timesteps non-uniformly
+                u = compute_density_for_timestep_sampling(
+                    weighting_scheme=args.weighting_scheme,
+                    batch_size=bsz,
+                    logit_mean=args.logit_mean,
+                    logit_std=args.logit_std,
+                    mode_scale=args.mode_scale,
+                )
+                indices = (u * noise_scheduler_copy.config.num_train_timesteps).long()
+                timesteps = noise_scheduler_copy.timesteps[indices].to(device=pixel_latents.device)
+
+                # Add noise according to flow matching.
+                sigmas = get_sigmas(timesteps, n_dim=pixel_latents.ndim, dtype=pixel_latents.dtype)
+                noisy_model_input = (1.0 - sigmas) * pixel_latents + sigmas * noise
```

Comment:
I thought we were using a different timestep sampling procedure and I suggested to have that as a default. Are we not doing that anymore?

Reply:
Do you mean to set the original sampling scheme as default? 
![image](https://github.com/user-attachments/assets/6a631c09-6cd4-4b87-b4c7-26dbcf6cfb36)
For the weighting schema i just copied from [here].(https://github.com/huggingface/diffusers/blob/48e36353d8cbf0322ec1ad0684b95d11f70af2de/examples/dreambooth/train_dreambooth_lora_flux.py#L1634C21-L1634C58)

Reply:
Yeah I meant to keep the `sigmoid` sampling as your default and let users configure it as we do in the other scripts.

Reply:
Could you please write it down briefly? I'm not sure how to edit it. It seems to me that if you use logit_normal, you should be using sigmoid?
![image](https://github.com/user-attachments/assets/b14214b1-9eb8-4f63-9e78-0d65af9665d1)

Reply:
Just need to change weighting_scheme from the default value to logit_normal?
![image](https://github.com/user-attachments/assets/cfdb3efa-78d1-4bfc-a7a6-be7986033ea2)

Reply:
Okay. But it depends on an std and mean. IIRC your scheme did `torch.randn()` and applied sigmoid right?

Reply:
Yes, this uses `torch.randn()` at first, but after given the examples you provided, I think this is the better solution for us.